ydb/core/kqp/gateway/kqp_metadata_loader.cpp

```diff
@@ -666,14 +659,25 @@ NThreading::TFuture<TTableMetadataResult> TKqpTableMetadataLoader::LoadIndexMeta
     auto loadIndexMetadataChecker =
         [ptr, result{std::move(loadTableMetadataResult)}, children](const NThreading::TFuture<void>) mutable {
             bool loadOk = true;
-            for (const auto& child : children) {
-                result.AddIssues(child.GetValue().Issues());
-                if (!child.GetValue().Success()) {
-                    loadOk = false;
+
+            const auto indexesCount = result.Metadata->Indexes.size();
+            result.Metadata->ImplTables.resize(indexesCount);
+            auto it = children.begin();
+            for (size_t i = 0; i < indexesCount; i++) {
+                for (const auto& _ : NTableIndex::GetImplTables(NYql::TIndexDescription::ConvertIndexType(
+                        result.Metadata->Indexes[i].Type))) {
+                    auto value = it++->ExtractValue();
+                    result.AddIssues(value.Issues());
+                    if (loadOk && (loadOk = value.Success())) {
+                        UpdateMetadataIfSuccess(result.Metadata->ImplTables[i], value);
+                    }
                 }
             }
+            Y_ASSERT(it == children.end());
```

Comment:
YQL_ENSURE?

Reply:
basically afaik it's most common practice to use it in KQP code, because we catch it and produce an error on request.

Reply:
Ok